src/securejoin.rs

```diff
@@ -139,14 +139,22 @@ async fn get_self_fingerprint(context: &Context) -> Result<Fingerprint> {
     Ok(key.dc_fingerprint())
 }
 
+// hack around the changed JSON accidentally used by an iroh upgrade, see #6518 for more details and for code snippet.
+// this hack is mainly needed to give ppl timme to upgrade and can be removed after some months (added 2025-02)
+pub(crate) fn fix_add_second_device_qr(qr: &str) -> String {
+    qr.replacen(",\"info\":{\"relay_url\":", ",\"relay_url\":", 1)
+        .replacen("\"]}}", "\"]}", 1)
```

Comment:
FYI you can use raw strings so that you don't need all these `\"`, e.g. `r#","info":{"relay_url":"#`. Rust-analyzer even has an assist "Rewrite as raw string" for that. But since it works like this, and since this code isn't supposed to stay in forever anyway, it's fine as-is.